### ISSUE TITLE
JAX 0.1.60 has exception on import for python3.5

### ISSUE THREAD

**Issue by brianwa84** [google/jax#2438]
```
  File "/tmpfs/pip_test_virtualenv/lib/python3.5/site-packages/jax/__init__.py", line 19, in <module>
    from jax.api import *
  File "/tmpfs/pip_test_virtualenv/lib/python3.5/site-packages/jax/api.py", line 31, in <module>
    from typing import Any, Callable, Collection, Optional, Sequence, Tuple, Union
ImportError: cannot import name 'Collection'
```

Looks likely related to https://github.com/google/jax/pull/2387

**Comment by hawkinsp** [comment]
Thanks for the report. I guess we'll avoid using `Collection` to stay Python 3.5-friendly. We should probably add testing for Python 3.5 as well.

Out of curiosity, do you have a strong reason to stay on Python 3.5? Would you object if we simply dropped compatibility sometime soon?

**Comment by brianwa84** [comment]
Not necessarily a strong reason. py35 is just the image we're running on
for internal OSS py3 testing.

Brian Patton | Software Engineer | bjp@google.com

**Comment by hawkinsp** [comment]
We talked over this, and we are thinking we will instead drop Python 3.5 compatibility.

One reason to do this is that NumPy is due to drop Python 3.5 compatibility on their next release (i.e., the window of support has already expired, per their deprecation policy.)
https://numpy.org/neps/nep-0029-deprecation_policy.html

**Comment by hawkinsp** [comment]
We're going to make one final release of `jax` that supports Python 3.5, and then drop Python 3.5 support for all future `jax` and `jaxlib` releases. Please update your CI build?

**Comment by hawkinsp** [comment]
I made a jax v0.1.61 release that fixes Python 3.5.

**Comment by brianwa84** [comment]
And I've upgraded to py36. All good, thx.

Brian Patton | Software Engineer | bjp@google.com
